### DISCUSSION TITLE
MWP Tools on Ubuntu

### DISCUSSION THREAD

**Discussion by synersignart** [stronnag/mwptools#100] [General]
Hi Stronnag I try installing the MWPtools today and run into problems 1st  being the instruction at the wiki  I was looking at is from the legacy installation

 https://github.com/stronnag/mwptools/wiki/Install-mwp-on-a-Windows-computer-for-Linux-noobs

which for some reason worked and installed

2nd by following the new instructions with the recent 2021 update I run into problems  
I got  Meason/meason build error : "Division works only with integers"
as for ninja install no such directory 

is there anyway I can build the MWPTools with the new procedure update for Ubuntu

and if possible to still have the legacy available for Git clone that way we have the options 

thanks

**Comment by stronnag** [comment]
* There is no dependency on `cmake`; the build tooling is `meson` and `ninja`.
* What version of Ubuntu ? You should not get an error on any _supported_ Ubuntu version.
* You can checkout any version you like. But it's probably not necessary; you should be able to:
  ```
  cd mwptools/src/mwp
  make && sudo make install
  ```
* It may be unnecessary as there is a mwptools_0.0.0.deb in the Release Area. 
* If you're still having problems, please provide:
  * The output of `cat /etc/lsb-release`
  * The full build log, not just the error, in order I can see where the problem might be.

**Reply by synersignart** [reply]
im using the image use the ubuntu version that was recommended by Nvidia Jetson Nano
 https://github.com/NVIDIA-AI-IOT/jetracer/blob/master/docs/software_setup.md

**Reply by stronnag** [reply]
> im using the image use the ubuntu version that was recommended by Nvidia Jetson Nano https://github.com/NVIDIA-AI-IOT/jetracer/blob/master/docs/software_setup.md

If you're not able to provide the information required, I'm unlikely to be able to help.

**Reply by synersignart** [reply]
hold on im trying to sort things out at the moment will update you

**Comment by stronnag** [comment]
Well it seems the device runs 18.04 (or some vendor modification thereof). That's too old; I only support LTS and latest. 
The `make` command may or may not work, as mwp may have dependencies on particular library functions that didn't exist in 18.04.

**Reply by synersignart** [reply]
https://github.com/stronnag/mwptools/wiki/Install-mwp-on-a-Windows-computer-for-Linux-noobs 
is this link still valid intruction. 

this is what i use on the Raspberry pi  Rasbian and a virtual machine of the Ubuntu desktop and it worked . the Jetson nano version  Ubuntu is having some issue

**Reply by stronnag** [reply]
Some it is; it now contains a warning about obsolescence.  I'm reluctant to butcher contributed articles, as it's hard enough to get them in the first place.

**Reply by stronnag** [reply]
Raspian is a sane distro, based on Debian stable, which is also a mwp supported OS.

**Comment by stronnag** [comment]
I no longer have any 18.04 virtual machines, so you'd need to find a commit (from `git log`) that is likely to (a) still work with `Makeflie` and (b) might work with 18.04 era compiler.
```
# This might be a candidate
git checkout 33978d4c6780cdbcdbabe271beaf2b2b3f8bfd05
make && sudo make install
```

**Reply by synersignart** [reply]
alright , also to build from source what else do i need .

**Reply by stronnag** [reply]
If you're using the old commit, `docs/ubuntu-deps.txt`; current master `docs/debian-ubuntu-dependencies.txt `

**Reply by synersignart** [reply]
for this recent release one

**Reply by synersignart** [reply]
or build it into a deb .

**Reply by stronnag** [reply]
https://blog.devgenius.io/how-to-build-debian-packages-from-meson-ninja-d1c28b60e709

**Comment by synersignart** [comment]
![viber_image_2021-10-18_10-48-34-293](https://user-images.githubusercontent.com/22503442/137661785-f693057a-baa4-44ed-af10-9f861fe29778.jpg)

**Comment by synersignart** [comment]
![viber_image_2021-10-18_10-49-39-088](https://user-images.githubusercontent.com/22503442/137661873-7aa12bc0-4909-40a2-a41a-a29572953979.jpg)

**Reply by synersignart** [reply]
the version I have was Ubuntu 18.04.6 LTS release 18.04 is this unsupported?

meson-log.txt

Build started at 2021-10-18T01:37:52.123568
Main binary: /usr/bin/python3
Python system: Linux
The Meson build system
Version: 0.45.1
Source dir: /home/jetson/projects/mwptools
Build dir: /home/jetson/projects/mwptools/build
Build type: native build
Project name: mwptools
Sanity testing C compiler: cc
Is cross compiler: False.
Sanity check compiler command line: cc /home/jetson/projects/mwptools/build/meson-private/sanitycheckc.c -o /home/jetson/projects/mwptools/build/meson-private/sanitycheckc.exe
Sanity check compile stdout:

-----
Sanity check compile stderr:

-----
Running test binary command: /home/jetson/projects/mwptools/build/meson-private/sanitycheckc.exe
Native C compiler: cc (gcc 7.5.0 "cc (Ubuntu/Linaro 7.5.0-3ubuntu1~18.04) 7.5.0")
Running compile:
Working directory:  /tmp/tmp5726_qe5
Command line:  valac /tmp/tmp5726_qe5/testfile.vala -C -C -o /tmp/tmp5726_qe5/output.obj 

Code:
 class MesonSanityCheck : Object { }
Compiler stdout:
 Compilation succeeded - 1 warning(s)

Compiler stderr:
 warning: --output and -o have no effect when -C or --ccode is set

Native Vala compiler: valac (valac 0.40.23)
Build machine cpu family: aarch64
Build machine cpu: aarch64
Program meson/post_install.sh found: YES (/home/jetson/projects/mwptools/meson/post_install.sh)
Configuring mwp-config.h using configuration

meson.build:12:0: ERROR: Non-existent build file 'meson/meson.build'

**Comment by stronnag** [comment]
meson is broken on your platform.
The minimum supported version is the current LTS 20.04

**Reply by synersignart** [reply]
I will try on Rasbian too to see if this works otherwise will use the LTS 20.04 but id really want to use those small boards like Raspberry and Jetsons for my applications

**Reply by stronnag** [reply]
Sure, it will work.
Just set 
```
export XDG_DATA_DIRS=/usr/share:$HOME/.local/share
```

in `~/.profile`, as it's ancient Debian 10.

![Screenshot-20211018171554-1136x978](https://user-images.githubusercontent.com/158229/137769788-d4369e4e-a060-4131-933e-7ab74e1487c0.png)

**Reply by synersignart** [reply]
thanks much appreciate it.  my feed back  to contribute is to do a separate wiki and git for the legacy Make file version and latest meson and Ninja version . which I did get confuse being on the same page.

and turn it to step by step guide from clean install .

cant help it that small boards like the raspberry and Jetson Nano sometimes runs of older 32bit operating systems.

cheers , wish i can contribute something too . given we are doing reproduction MWC boards which could make use of updated ground station.

**Comment by stronnag** [comment]
The wiki already:
* Makes clear that the legacy contributed documentation is obsolete
* Provides a link to one-stop installation script

Good luck with the MWC boards; in theory mwp still supports them.

**Reply by synersignart** [reply]
for Debian 10 
 lets do it again from start

cd #home
mkdir Projects
cd projects
git clone https://github.com/stronnag/mwptools.git
cd mwptools
export XDG_DATA_DIRS=/usr/share:$HOME/.local/share

this is where im confuse
and if there are dependencies lacking where do i find them

**Reply by stronnag** [reply]
Just following the [**documented** procedure](https://github.com/stronnag/mwptools/wiki/Building-with-meson-and-ninja#easy-first-time-install-on-debian-and-ubuntu) worked for me.

**Reply by synersignart** [reply]
![error](https://user-images.githubusercontent.com/22503442/137943792-7a32879b-f5f1-47c0-9318-951dab3a7672.png)
ok did i miss anything the whole thing install on raspbian  following the first time build script but end up with a broken extension  how to fix?

https://raw.githubusercontent.com/stronnag/mwptools/master/docs/deb-install.sh

excuse me for running into error was testing different OS to see what works and what dont and need to understand procedure each one of them

for now 
Ubuntu 20.04 works no issue from instruction
Nvidia Ubuntu 18.04 Meson Error
Raspbian desktop broken application extension (update - issue goes away after reboot and the application manage to execute)

**Reply by stronnag** [reply]
No such issue here.

**Reply by synersignart** [reply]
well the issue goes away after reboot. hmm weird. must be something when running on virtual machine. will have to note it

**Comment by stronnag** [comment]
* Raspian is not a supported OS (it's Debian 10, stable is Debian 11).
* If a file is broken, show me the file contents, not a image
* Use a supported OS version.

**Reply by synersignart** [reply]
sadly wasnt able to recreate it since after reboot everything works   must to be because it was on the root folder which require to load on boot

correction its Debian Buster with Raspberry Pi Desktop 

ok so to conclude on my end 

Ubuntu 20.04 works no issue from instruction  (prefix=/usr)

Nvidia Ubuntu 18.04 Meson Error

Debian Buster with Raspberry Pi Desktop -  require reboot as the application is in the root folder (any other better location to put it in other than (prefix ~/.local )


https://www.raspberrypi.com/software/raspberry-pi-desktop/
![works](https://user-images.githubusercontent.com/22503442/137951862-c9604465-ed38-4c87-817a-6acd8eab5ffe.png)

**Comment by synersignart** [comment]
Cheers, Stronnag ,  sorry I have been difficult at first Id say id understand if instruction was laid out in sequences as how i understand it.  so far this is what worked on my end 


----------------------------------------------------------------------------------------------------
FOR UBUNTU 20

cd #home

sudo apt update && sudo apt full-upgrade && \
  sudo apt $CONFIRM install libgdl-3-dev \
    libchamplain-gtk-0.12-dev \
    libchamplain-0.12-dev \
    libclutter-1.0-dev \
    libclutter-gtk-1.0-dev \
    libgtk-3-dev \
    valac \
    pkg-config \
    git build-essential \
    meson \
    ninja-build \
    libbluetooth-dev \
    libespeak-dev \
    libgudev-1.0-dev \
    libgstreamer1.0-dev \
    libvte-2.91-dev \
    libncurses5-dev \
    golang-go \
    ruby ruby-json \
    libspeechd-dev flite flite1-dev libmosquitto-dev \
    gnuplot ruby-nokogiri unzip



mkdir Projects

cd projects

git clone https://github.com/stronnag/mwptools.git 

cd mwptools

meson build --prefix=/usr

cd build

ninja && ninja install


----------------------------------------------------------------------------------------------------

FOR DEBIAN 10

cd #home

sudo apt update && sudo apt full-upgrade && \
  sudo apt $CONFIRM install libgdl-3-dev \
    libchamplain-gtk-0.12-dev \
    libchamplain-0.12-dev \
    libclutter-1.0-dev \
    libclutter-gtk-1.0-dev \
    libgtk-3-dev \
    valac \
    pkg-config \
    git build-essential \
    meson \
    ninja-build \
    libbluetooth-dev \
    libespeak-dev \
    libgudev-1.0-dev \
    libgstreamer1.0-dev \
    libvte-2.91-dev \
    libncurses5-dev \
    golang-go \
    ruby ruby-json \
    libspeechd-dev flite flite1-dev libmosquitto-dev \
    gnuplot ruby-nokogiri unzip


mkdir -p ~/.local/bin

git clone https://github.com/stronnag/mwptools.git 

cd mwptools

meson build --buildtype=release --strip --prefix ~/.local

cd build

ninja && ninja install

**Reply by synersignart** [reply]
@ stronnag , I really Appreciate your contribution to the community  im also deploying Development board base flight controllers mainly for schools  and its really big help on for the STEM community.
